[DOC] Add set processor documentation (#5989)
opensearch-project/documentation-website: _ingest-pipelines/processors/set.md

---
layout: default
title: Set
parent: Ingest processors
nav_order: 240
---

# Set processor

> The `set` processor adds or updates fields in a document. It sets one field and associates it with the specified value. If the field already exists, its value is replaced with the provided one.

Author: ```suggestion
The `set` processor adds or updates fields in a document. It sets one field and associates it with the specified value. If the field already exists, its value is replaced with the provided one, unless the override parameter is set to false. When override is false and the specified field exists, the value of the field will remain unchanged.
```

Author: @gaobinlong Please confirm the change accurately reflects your comment. Thank you.

Reviewer: Edited suggestion

The following is the syntax for the `set` processor:

```json
{
  "description": "...",
  "processors": [
    {
      "set": {
        "field": "new_field",
        "value": "some_value"
      }
    }
  ]
}
```
{% include copy-curl.html %}

## Configuration parameters

The following table lists the required and optional parameters for the `set` processor.

Parameter | Required/Optional | Description |
|-----------|-----------|-----------|
`field` | Required | The name of the field to be set or updated. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets).
`value` | Required | The value to be assigned to the field. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets).

> `overrideEnabled` | Optional | A Boolean flag that determines whether the processor should override the exisitng value of the field.

Author: ```suggestion
`override` | Optional | A Boolean flag that determines whether the processor should override the existing value of the field.
```

> `ignoreEmptyValue` | Optional | A Boolean flag that determines whether the processor should ignore empty values.

Author: ```suggestion
`ignore_empty_value` | Optional | A Boolean flag that determines whether the processor should ignore `null` values or empty strings. Default is `false`.
```

`description`  | Optional  | A description of the processor's purpose or configuration.
`if` | Optional | Specifies to conditionally execute the processor.

> `ignore_failure` | Optional | Specifies to ignore failures for the processor. See [Handling pipeline failures]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/).

Reviewer: ```suggestion
`ignore_failure` | Optional | Specifies to ignore processor failures. See [Handling pipeline failures]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/).
```

```json
PUT _ingest/pipeline/set-pipeline
{
  "description": "Adds a new field 'new_field' with the value 'some_value'",
  "processors": [
    {
      "set": {
        "field": "new_field",
        "value": "some_value"
      }
    }
  ]
}
```
{% include copy-curl.html %}

### Step 2 (Optional): Test the pipeline

It is recommended that you test your pipeline before you ingest documents.
{: .tip}

To test the pipeline, run the following query:

```json
POST _ingest/pipeline/set-pipeline/_simulate
{
  "docs": [
    {
      "_source": {
        "existing_field": "value"
      }
    }
  ]
}
```
{% include copy-curl.html %}

#### Response

The following example response confirms that the pipeline is working as expected:

```json
{
  "docs": [
    {
      "doc": {
        "_index": "_index",
        "_id": "_id",
        "_source": {
          "existing_field": "value",
          "new_field": "some_value"
        },
        "_ingest": {
          "timestamp": "2024-05-30T21:56:15.066180712Z"
        }
      }
    }
  ]
}
```
{% include copy-curl.html %}

### Step 3: Ingest a document 

The following query ingests a document into an index named `testindex1`:

```json

> POST testindex1/_doc

Author: ```suggestion
POST testindex1/_doc?pipeline=set-pipeline
```

---
layout: default
title: Set
parent: Ingest processors
nav_order: 240
---

# Set processor

The `set` processor is used to <explain what is used to do>.

The following is the syntax for the `set` processor:

```json
<insert syntax example>
```
{% include copy-curl.html %}

## Configuration parameters

The following table lists the required and optional parameters for the `set` processor.

Parameter | Required/Optional | Description |

> |-----------|-----------|-----------|

Author: @gaobinlong Please confirm all parameters are correct and listed for this processor.
